Remove port number in URL for url_norm

thomasegense opened the pull request:
This closes: https://github.com/ukwa/webarchive-discovery/issues/284 Additional minor clean up.

Created CHANGES.MD . Added recent fix by Leslie that was not mentioned.
Minor cleanup.
Added the default configuration config3.xml for the warc-indexer.

thomasegense commented:
The new conf/config3.xml is the suggested default configuration for using the warc-indexer. A config3.xml is required.
When using SolrWayback and good default config3.xml is also added.
For users only checking out the warc-indexer project they will need a resonable config. All the configs in the unittest resources are not good defaults.
@VictorHarbo

kesj-kb commented on src/main/java/uk/bl/wa/util/Normalisation.java:
```diff
@@ -307,4 +320,34 @@ private static byte[] base16ToBytes(String hex) {
+        
+        if (!inputUrl.startsWith("http://") || inputUrl.toLowerCase().startsWith("https://")) {
+            return inputUrl;
+        }
```
If a URL starts with "https://", then it definitely does not start with "http://".
So, the condition !inputUrl.startsWith("http://") will already evaluate to true for all https:// URLs.
This means the second condition is redundant in this context.
Maybe include a null check?

thomasegense commented on CHANGES.md:
```diff
@@ -0,0 +1,150 @@
+
+**NOTE** Generally, we only add terms to the Solr schema, so it should usually be compatible with previous versions (i.e. clients should be able to query across both without modification). However, there are been a small number of fixes which unfortunately required breaking changes you may need to be aware of or work-around. e.g. [hash becomes single-valued](https://github.com/ukwa/webarchive-dis ...
```
Copied from old changelog... Will clear later what happened.

thomasegense commented:
@VictorHarbo  Just awaiting you now. I fixed the remotePort method and added javadoc so it will work as a standalone method also and not require addition normalization before called.
